lxd/storage/backend_lxd.go

```diff
@@ -3163,10 +3163,20 @@ func (b *lxdBackend) EnsureImage(fingerprint string, op *operations.Operation) e
 		}
 	}
 
-	// Validate config and create database entry for new storage volume.
-	err = VolumeDBCreate(b, project.Default, fingerprint, "", drivers.VolumeTypeImage, false, volConfig, time.Now().UTC(), time.Time{}, contentType, false)
+	// For drivers which support both block mode and non-block mode, there should always only be a
+	// single database entry even if two different volumes exist on disk.
+	// We therefore check for an existing entry, and only create one if not present.
+	_, err = VolumeDBGet(b, project.Default, fingerprint, drivers.VolumeTypeImage)
```

Comment:
There is already `imgDBVol` var which if non-nil indicates if the record exists already or not without having to run additional queries here.